pkgs/os-specific/linux/systemd/0015-path-util.h-add-placeholder-for-DEFAULT_PATH_NORMAL.patch

```diff
@@ -19,8 +19,8 @@ index 22d3632e6e..1e8bbb242b 100644
  
 -#define DEFAULT_PATH_NORMAL PATH_SBIN_BIN("/usr/local/") ":" PATH_SBIN_BIN("/usr/")
 -#define DEFAULT_PATH_NORMAL_NULSTR PATH_SBIN_BIN_NULSTR("/usr/local/") PATH_SBIN_BIN_NULSTR("/usr/")
-+#define DEFAULT_PATH_NORMAL "@defaultPathNormal@"
-+#define DEFAULT_PATH_NORMAL_NULSTR "@defaultPathNormal@\0"
++#define DEFAULT_PATH_NORMAL "@defaultPathNormal@" ":" PATH_SBIN_BIN("/run/current-system/sw/")
```

Comment:
from `man systemd.service`:

> If the command is not a full (absolute) path, it will be resolved to a full path using a fixed search path determined at compilation time. Searched directories include /usr/local/bin/, /usr/bin/, /bin/ on systems using split /usr/bin/ and /bin/ directories, and their sbin/ counterparts on systems using split bin/ and sbin/. It is thus safe to use just the executable name in case of executables located in any of the "standard" directories, and an absolute path must be used in other cases. Using an absolute path is recommended to avoid ambiguity. Hint: this search path may be queried using systemd-path search-binaries-default.

```
$ systemd-path search-binaries-default 
/nix/store/wm0fbn9fg03b1xmaqs6zxxp66yqfp0qh-systemd-252.4/bin/
```

So, while the systemd documentation seems to indicate NixOS is already kind of out of line, and therefore we can kind of do what we want, I do think it would be *very out of character* for NixOS to allow service units to name executables without their absolute paths. So I do not support this change to this systemd patch.